wsmaster/che-core-api-factory-gitlab-common/src/main/java/org/eclipse/che/api/factory/server/gitlab/AbstractGitlabUrlParser.java

```diff
@@ -112,7 +112,8 @@ private boolean isApiRequestRelevant(String repositoryUrl) {
         // belongs to Gitlab.
         gitlabApiClient.getOAuthTokenInfo("");
       } catch (ScmUnauthorizedException e) {
-        return true;
+        // the error message is a JSON if it is a response from Gitlab.
+        return e.getMessage().startsWith("{");
```

Comment:
Do we really need this changes in context of azure?

Reply:
Yes, we need to specify the unauthorized exception as Azure also returns an unauthorized exception in this case.

Reply:
what is the response from Azure in this case?